ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -0,0 +1,565 @@
+    netcdf_file_obj   = Dataset(nc_file_list[0], 'r')
+    site_code         = netcdf_file_obj.site_code
+    deployment_code   = netcdf_file_obj.deployment_code
+    input_netcdf_name = os.path.basename(nc_file_list[0])
+    pattern           = re.compile("^(IMOS_.*)_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_(.*)_FV01_(.*)_END")
+    match_group       = pattern.match(input_netcdf_name)
+
+    time_values = transform_datetime64_datetime(df.columns.values)
+    time_start  = min(time_values).strftime('%Y%m%dT%H%M%SZ')
+    time_end    = max(time_values).strftime('%Y%m%dT%H%M%SZ')
+
+    output_netcdf_name = '%s_T_%s_%s_FV02_%s_gridded_END-%s.nc' % (match_group.group(1), time_start,
+                                                                    site_code, deployment_code, time_end)
+
+    netcdf_file_obj.close()
+    return output_netcdf_name
+
+def transform_datetime64_datetime(datetime64_object):
+    """ The pandas date object is turned into a datetime64 type. This creates an issue when
+    doing a normal datetime operation which is fixed by running this function """
+    # http://stackoverflow.com/questions/13703720/converting-between-datetime-timestamp-and-datetime64
+    ns          = 1e-9 # nanoseconds
+    time_values = [datetime.utcfromtimestamp(x.astype(int) * ns) for x  in datetime64_object]
+    return time_values
+
+def generate_fv02_netcdf(df, nc_file_list):
+    """ generated the FV02 temperature gridded product netcdf file """
+    tmp_netcdf_dir          = tempfile.mkdtemp()
+    output_netcdf_file_path = os.path.join(tmp_netcdf_dir, generate_fv02_filename(df, nc_file_list))
+
+    input_netcdf_obj               = Dataset(nc_file_list[0], 'r')
+    output_netcdf_obj              = Dataset(output_netcdf_file_path, "w", format="NETCDF4")
+    output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # read gatts from input, add them to output. Some gatts will be overwritten
+    input_gatts     = input_netcdf_obj.__dict__.keys()
+    gatt_to_dispose = ['author', 'toolbox_input_file', 'file_version', 'file_version_quality_control', 'quality_control_set',
+                       'CoordSysBuilder_', 'netcdf_filename', 'metadata', 'instrument_serial_number',
+                       'instrument_nominal_depth', 'compliance_checker_version', 'compliance_checker_last_updated',
+                       'geospatial_vertical_min', 'geospatial_vertical_max', 'featureType']
+
+    for gatt in input_gatts:
+        if gatt not in gatt_to_dispose:
+            setattr(output_netcdf_obj, gatt, getattr(input_netcdf_obj, gatt))
+
+    comment = 'comment: The following files have been used to generate the gridded product:\n%s' % " \n".join([os.path.basename(x) for x in nc_file_list])
+    setattr(output_netcdf_obj, 'comment', comment)
+    setattr(output_netcdf_obj, 'temporal_resolution', get_frequency_step_in_deployment(nc_file_list)[0] )
+    setattr(output_netcdf_obj, 'vertical_resolution', 1 )
```

Comment:
Is it because the resolution values are sometimes more than nine quintillion, two hundred twenty-three quadrillion, three hundred seventy-two trillion, thirty-six billion, eight hundred fifty-four million, seven hundred seventy-five thousand, eight hundred seven?

```bash
$ python -c 'import sys; print(sys.maxint)'
9223372036854775807
```
https://docs.python.org/2/library/stdtypes.html#typesnumeric
> Plain integers (also just called integers) are implemented using long in C, which gives them at least 32 bits of precision (sys.maxint is always set to the maximum plain integer value for the current platform, the minimum value is -sys.maxint - 1). Long integers have unlimited precision.